content/blog/langchain-basic-memory-types.md

```diff
@@ -0,0 +1,145 @@
++++
+title = "LangChain Basic Memory Types: Buffer, BufferWindow, TokenBuffer and Summary"
+slug = "langchain-basic-memory-types"
+date = 2024-06-30T14:49:39+05:30
+image = "/images/2024/langchain-basic-memory-types.png"
+draft = false
+authors = ["Syed Mohd Mehndi"]
+description = "LangChain Basic Memory Types: Buffer, BufferWindow, TokenBuffer and Summary"
+tags = ["AI", "Langchain"]
+categories = ["AI", "Langchain"]
+type = ""
++++
+
+LangChain provides a powerful way to manage conversational memory in chatbots and conversational agents. Among the various memory types it supports, four key ones stand out: ConversationBufferMemory, ConversationBufferWindowMemory, ConversationTokenBufferMemory and ConversationSummaryMemory. Each of these memory types serves a unique purpose and can be used in different scenarios based on the requirements of the conversation. In this blog, we'll explore these memory types and provide short examples to highlight their differences.
+
+### ConversationBufferMemory
+
+ConversationBufferMemory stores the entire conversation history. This is useful when you need to retain every part of the conversation for context.
+
+#### Example:
+
+```python
+from langchain.memory import ConversationBufferMemory
+
+# Initialize buffer memory
+memory = ConversationBufferMemory()
+
+# Add conversation turns
+memory.add_message("User", "Hello, how are you?")
+memory.add_message("Assistant", "I'm good, thank you! How can I assist you today?")
+memory.add_message("User", "Can you tell me about the weather?")
+
+# Retrieve memory
+print(memory.retrieve())
+```
+
+#### Output:
+
+```vbnet
```

Comment:
Any specific reason for using vbnet foor highlighting here?

Reply:
have changed this to ```plaintext now to avoid any syntax highlighting